### PR TITLE
[Agent Management] Add agent management-related alerts

### PR DESCRIPTION
**Pull request by spartan0x117**
#### PR Description
Adds some `warning` and `critical` alerts for agent management-related errors.

- [ ] CHANGELOG updated
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

**Review by marctc — APPROVED** [review]

File: production/grafana-agent-mixin/alerts.libsonnet [hunk 1]
```diff
@@ -235,6 +235,151 @@ local _config = config._config;
...
           },
         ],
+      },
+      {
+        name: 'GrafanaAgentManagement',
+        rules: [
+          {
+            alert: 'AgentManagementBadAPIRequests',
+            expr: |||
+              100 * sum(rate(agent_remote_config_fetches_total{status_code!~"2.."}[5m])) by (%(group_by_cluster)s)
```
Comment (jcreixell) [inline production/grafana-agent-mixin/alerts.libsonnet:245]: shall this alert fire if we get `3..` responses?

Comment (jcreixell) [inline production/grafana-agent-mixin/alerts.libsonnet:245]: I think it is probably safer to use 10m, I could see agents having multi-minute poll intervals to reduce request rates

Comment (jcreixell) [reply production/grafana-agent-mixin/alerts.libsonnet:245]: probably safer to use `"(4|5).."`

**Review by jcreixell — APPROVED** [review]
please see my comments, other than that LGTM

File: production/grafana-agent-mixin/alerts.libsonnet [hunk 2]
```diff
@@ -242,12 +242,12 @@ local _config = config._config;
...
             expr: |||
-              100 * sum(rate(agent_remote_config_fetches_total{status_code!~"2.."}[5m])) by (%(group_by_cluster)s)
+              100 * sum(rate(agent_remote_config_fetches_total{status_code="(4|5).."}[10m])) by (%(group_by_cluster)s)
                 /
-              sum(rate(agent_remote_config_fetches_total[5m])) by (%(group_by_cluster)s)
+              sum(rate(agent_remote_config_fetches_total[10m])) by (%(group_by_cluster)s)
                 > 5
             ||| % _config,
-            'for': '10m',
+            'for': '20m',
```
Comment (jcreixell) [inline production/grafana-agent-mixin/alerts.libsonnet:250]: AFAIK we don't need to change this when increasing the rate interval (it just means it looks back a longer time frame for each data point, but the alert duration, meaning how much we wait before alerting, is not affected)

**Review by jcreixell — APPROVED** [review]
Please see my comment, but overall LGTM

**Pull request merged**